api/Updated-Metrics.md

```diff
@@ -0,0 +1,394 @@
+```python
+>>> result = flm.group_summary(skm.accuracy_score, y_true, y_pred, sensitive_features=A_1)
+>>> result.metric_
+<class 'function'>
+>>> result.overall
+0.4
+>>> result.by_groups
+B      0.6536
+C      0.2130
+Name: accuracy_score dtype: float64
+>>> print(type(result.by_groups))
+<class 'pandas.core.series.Series'>
+```
+Constructing the name of the Series could be an issue.
+In the example above, it is the name of the underlying metric function.
+Something as short as the `__name__` could end up being ambiguous, but using the `__module__` property to disambiguate might not match the user's expectations:
+```python
+>>> import sklearn.metrics as skm
+>>> skm.accuracy_score.__name__
+'accuracy_score'
+>>> skm.accuracy_score.__qualname__
+'accuracy_score'
+>>> skm.accuracy_score.__module__
+'sklearn.metrics._classification'
+```
+We are seeing some of the actual internal structure here of SciKit-Learn, and the user might not be expecting that.
+
+We would continue to provide convenience wrappers such as `accuracy_score_group_summary` for users, and support passing through arguments along with `indexed_params`.
+There is little advantage to the change at this point.
+This will change in the next section.
+
+## Obtaining Scalars
+
+### Existing Syntax
+
+We provide methods for turning the `Bunch`es returned from `group_summary()` into scalars:
+```python
+>>> difference_from_summary(result)
+0.4406
+>>> ratio_from_summary(result)
+0.3259
+>>> group_max_from_summary(result)
+0.6536
+>>> group_min_from_summary(result)
+0.2130
+```
+We also provide wrappers such as `accuracy_score_difference()`, `accuracy_score_ratio()` and `accuracy_score_min()` for user convenience.
+
+One point which these helpers lack (although it could be added) is the ability to select alternative values for measuring the difference and ratio.
+For example, the user might not be interested in the difference between the maximum and minimum, but the difference from the overall value.
+Or perhaps the difference from a particular group.
+
+### Proposed Change
+
+The `GroupedMetric` object would have methods for calculating the required scalars.
+First, let us consider the differences.
+
+We would provide operations to calculate differences in various ways (all of these results are a Pandas Series):
+```python
+>>> result.differences()
+B      0.0
+C      0.4406
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min')
+B     -0.4406
+C      0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min', abs=True)
+B      0.4406
+C      0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall')
+B     -0.2436
+C      0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall', abs=True)
+B      0.2436
+C      0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='group', group='C', abs=True)
+B      0.4406
+C      0.0
+Name: TBD dtype: float64
+```
```

Comment:
I think that you are taking the opposite difference than I had in mind. You should take the difference between the column containing group-level metrics and some scalar described by "relative_to".

Reply:
I may well have done this the other way around. I can see that `x-x0` might make more sense than `x0-x`.

Reply:
A quick survey (N=3) tells me that the problem with `relative_to` is that people may automatically assume it refers to the absolute difference. I.e. a sentence like "For each group, we compute the difference in accuracy relative to the maximum accuracy score" translates to |x_max - x_group|. One way to tackle this would be to make the arguments more explicit, i.e. use something like `subtrahend` and `denominator`, but then we'd have a different keyword argument for almost the same thing, which may not be nice. We can also try to make the meaning very clear in the docs by including the formula, but I'm not sure whether it's naive to assume that people actually read docs :P

Reply:
My N=1 survey saw `subtrahend` and went scuttling off to a dictionary to see if it meant what I thought it did :-) And the problem with `denominator` for the ratios is that in some of the cases I outline, the selected value will be the _numerator_. The best I can think up right now is `fulcrum` but I'm not fond of that either.

The name and the behaviour are two separate things, though. Using the current nomenclature, if `relative_to` selects `x0` then does `differences()` calcualte `x-x0` or `x0-x`. Now that I write it down, I'm more used to seeing the former (in which case the default would be `min` as @MiroDudik  states, giving all values to be positive regardless of `abs=`).